### PR TITLE
chore: rename project to example-storefront

### LINKED ISSUE #476: Rename this project/repo
Type: **breaking|critical|major**

**Describe the bug**
The `reaction-next-starterkit` name came about mostly as an inside joke that made it all the way to the repo. The name is extremely confusing to not only the community but within the Reaction team. 

**To Reproduce**
1. Go to the `reaction-next-starterkit` repo.
2. Read the project name.
3. You should now be confused.

**Expected behavior**
A clear project/repo name that accurately conveys the reference/example intent of this project.

**Additional context**
I see two deliverables here before we actually change the projects name.
1. [x] Pick a new name: Current front runner `example-storefront`
2. [ ] Audit other Reaction repos that may reference this project by name. (i.e., `reaction-platform`)

### PR DESCRIPTION
**Pull request by nnnnat**
Resolves #476 
Impact: **breaking|critical|major**
Type: *chore**

## Issue
This project was long due for a rename as the "starterkit" name caused some confusion. 

## Solution
This PR get's this project ready for the official 2.0 release by changing the project/repo name to "example-storefront", updating configs to use the `graphql-beta` endpoint and some copy/docs updates.

## Testing
n/a

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,11 @@
-# Reaction Storefront Next.js Starter Kit
+# Example Storefront
 [![FOSSA Status](https://app.fossa.io/api/projects/git%2Bgithub.com%2Freactioncommerce%2Freaction-next-starterkit.svg?type=shield)](https://app.fossa.io/projects/git%2Bgithub.com%2Freactioncommerce%2Freaction-next-starterkit?ref=badge_shield)
```
Comment (nnnnat) [inline README.md]: Not sure how these badges work and how to update it, looking for some guidance

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "reaction-next-starterkit",
-  "version": "2.0.0-rc.9",
+  "name": "example-storefront",
+  "version": "2.0.0",
```
Comment (nnnnat) [inline package.json]: @spencern not sure if I should change this or not

Comment (aldeed) [reply package.json]: Changing the name is fine but I don't know that we should change the version yet. That's part of the release checklist.

File: README.md (hunk 1 above)
Comment (aldeed) [reply README.md]: Docs: https://docs.fossa.com/docs/automatic-updates#section-adding-a-fossa-badge-to-documentation-readme

Guessing you'll want to rename to match the repo rename but not 100% sure.

**Pull request merged**